app/connector/kafka/src/main/java/io/syndesis/connector/kafka/KafkaConnector.java

```diff
@@ -15,40 +15,85 @@
  */
 package io.syndesis.connector.kafka;
 
+import java.util.HashMap;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import io.syndesis.connector.kafka.service.KafkaBrokerService;
 import io.syndesis.connector.support.util.ConnectorOptions;
 import io.syndesis.connector.support.util.KeyStoreHelper;
-
+import io.syndesis.integration.component.proxy.ComponentDefinition;
 import io.syndesis.integration.component.proxy.ComponentProxyComponent;
-import io.syndesis.integration.component.proxy.ComponentProxyCustomizer;
+import org.apache.camel.Component;
+import org.apache.camel.component.kafka.KafkaComponent;
 import org.apache.camel.component.kafka.KafkaConfiguration;
-
 import org.apache.camel.util.jsse.KeyManagersParameters;
 import org.apache.camel.util.jsse.KeyStoreParameters;
 import org.apache.camel.util.jsse.SSLContextParameters;
 import org.apache.camel.util.jsse.TrustManagersParameters;
-
 import org.apache.commons.beanutils.PropertyUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.kafka.common.config.SaslConfigs;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import static io.syndesis.connector.kafka.service.KafkaBrokerService.OAUTHBEARER;
+import static io.syndesis.connector.kafka.service.KafkaBrokerService.PLAIN;
 
-public class KafkaConnectionCustomizer implements ComponentProxyCustomizer {
```

Comment:
I'm still not sure why the customizer approach doesn't work here, is how secrets are fetched in the REST Swagger connector, for example fetching password here:

https://github.com/syndesisio/syndesis/blob/78d0ffb36cd15810b01aab864f6878aecb8e2f5f/app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/auth/basic/Basic.java#L33

Leads to:

https://github.com/syndesisio/syndesis/blob/78d0ffb36cd15810b01aab864f6878aecb8e2f5f/app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/Configuration.java#L77

Password being secret:

https://github.com/syndesisio/syndesis/blob/78d0ffb36cd15810b01aab864f6878aecb8e2f5f/app/connector/rest-swagger/src/main/resources/META-INF/syndesis/connector/rest-swagger.json#L171

I wonder if this is getting more complex than it needs to be.